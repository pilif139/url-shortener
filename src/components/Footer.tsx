'use client'

import {FaGithub} from "react-icons/fa";
import { FaLinkedin } from "react-icons/fa";
import Link from "next/link";
import {motion} from "framer-motion";
import DarkModeToggler from "@/components/DarkModeToggler";
import { Toaster } from "@/components/ui/sonner"

const linkAnimation = {
    scale: [1, 1.1, 1],
    rotate: [0, 10, -10, 0],
    transition: {
        duration: 0.5,
    }
}

export default function Footer(){
  return (
      <div
          className="relative bottom-0 flex justify-center text-5xl">
        <Toaster position="bottom-left"/>
        <motion.div
            initial={{opacity: 0, y: 50}}
            animate={{opacity: 1, y: 0}}
            transition={{duration: 0.5}}
            className="relative bottom-0 flex justify-between gap-3 p-5 text-5xl">
          <motion.div whileHover={linkAnimation}>
            <Link
                href={"https://github.com/pilif139"} target="_blank"
                className="transition duration-500 hover:text-gray-500">
              <FaGithub/>
            </Link>
          </motion.div>
          <motion.div whileHover={linkAnimation}>
            <Link href={"https://www.linkedin.com/in/filip-kasperski-b80a8b310/"} target="_blank"
                  className="transition duration-500 hover:text-gray-500">
              <FaLinkedin/>
            </Link>
          </motion.div>
        </motion.div>
        <div className="text-xl">
          <DarkModeToggler/>
        </div>
      </div>
  )
}