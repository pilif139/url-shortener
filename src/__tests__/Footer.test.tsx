import Footer from "@/components/Footer";
import {render, screen} from "@testing-library/react";

describe("Footer", ()=>{
  it("should render correctly", ()=>{
    const {container} = render(<Footer/>)
    expect(container).toMatchSnapshot();
  })

  it("should render links to socials", ()=>{
    render(<Footer/>)
    const githubLink = screen.getByLabelText(/github link/i);
    const linkedinLink = screen.getByLabelText(/linkedin link/i);
    expect(githubLink).toHaveAttribute("href", "https://github.com/pilif139");
    expect(linkedinLink).toHaveAttribute("href", "https://www.linkedin.com/in/filip-kasperski-b80a8b310/");
  });

    it("should render dark mode toggler", ()=>{
        render(<Footer/>)
        expect(screen.getByRole("button")).toBeInTheDocument();
    })
})