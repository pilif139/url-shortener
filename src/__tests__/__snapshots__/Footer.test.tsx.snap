// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should render correctly 1`] = `
<div>
  <footer
    class="relative bottom-0 flex justify-center text-5xl"
  >
    <div
      class="relative bottom-0 flex justify-between gap-3 p-5 text-5xl"
      style="opacity: 0; will-change: opacity,transform; transform: translateY(50px);"
    >
      <div>
        <a
          aria-label="Github Link"
          class="transition duration-500 hover:text-gray-500"
          href="https://github.com/pilif139"
          target="_blank"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 496 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"
            />
          </svg>
        </a>
      </div>
      <div>
        <a
          aria-label="Linkedin Link"
          class="transition duration-500 hover:text-gray-500"
          href="https://www.linkedin.com/in/filip-kasperski-b80a8b310/"
          target="_blank"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 448 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M416 32H31.9C14.3 32 0 46.5 0 64.3v383.4C0 465.5 14.3 480 31.9 480H416c17.6 0 32-14.5 32-32.3V64.3c0-17.8-14.4-32.3-32-32.3zM135.4 416H69V202.2h66.5V416zm-33.2-243c-21.3 0-38.5-17.3-38.5-38.5S80.9 96 102.2 96c21.2 0 38.5 17.3 38.5 38.5 0 21.3-17.2 38.5-38.5 38.5zm282.1 243h-66.4V312c0-24.8-.5-56.7-34.5-56.7-34.6 0-39.9 27-39.9 54.9V416h-66.4V202.2h63.7v29.2h.9c8.9-16.8 30.6-34.5 62.9-34.5 67.2 0 79.7 44.3 79.7 101.9V416z"
            />
          </svg>
        </a>
      </div>
    </div>
    <div
      class="text-xl"
    >
      <button
        class="absolute right-7 bottom-7 rounded-3xl bg-indigo-500 p-5 text-white transition-colors scale-[1.12] dark:bg-gray-900"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256 160c-52.9 0-96 43.1-96 96s43.1 96 96 96 96-43.1 96-96-43.1-96-96-96zm246.4 80.5l-94.7-47.3 33.5-100.4c4.5-13.6-8.4-26.5-21.9-21.9l-100.4 33.5-47.4-94.8c-6.4-12.8-24.6-12.8-31 0l-47.3 94.7L92.7 70.8c-13.6-4.5-26.5 8.4-21.9 21.9l33.5 100.4-94.7 47.4c-12.8 6.4-12.8 24.6 0 31l94.7 47.3-33.5 100.5c-4.5 13.6 8.4 26.5 21.9 21.9l100.4-33.5 47.3 94.7c6.4 12.8 24.6 12.8 31 0l47.3-94.7 100.4 33.5c13.6 4.5 26.5-8.4 21.9-21.9l-33.5-100.4 94.7-47.3c13-6.5 13-24.7.2-31.1zm-155.9 106c-49.9 49.9-131.1 49.9-181 0-49.9-49.9-49.9-131.1 0-181 49.9-49.9 131.1-49.9 181 0 49.9 49.9 49.9 131.1 0 181z"
          />
        </svg>
      </button>
    </div>
  </footer>
</div>
`;
